packages/video_player/video_player_platform_interface/lib/video_player_platform_interface.dart

```diff
@@ -102,6 +102,11 @@ abstract class VideoPlayerPlatform extends PlatformInterface {
   Future<void> setMixWithOthers(bool mixWithOthers) {
     throw UnimplementedError('setMixWithOthers() has not been implemented.');
   }
+
+  /// Sets additional options on web
+  Future<void> setWebOptions(int textureId, VideoPlayerWebOptions controls) {
+    throw UnimplementedError('setWebOptions() has not been implemented.');
+  }
```

Comment:
@stuartmorgan WRT the new API. We just landed another way of adding "web-only API" to a plugin, when people want to render a sign-in button for the web, like this:

* https://github.com/flutter/packages/blob/eabaaccd0580a2e9f171f048a68070ea6f63a566/packages/google_sign_in/google_sign_in/example/lib/src/sign_in_button/web.dart#L11-L15

Even though extending that API (and an associated configuration object) requires only changing the `_web` package, **using** the new code requires programmers to conditionally import the web package and call its web-only method; I think the resulting code is much more cumbersome.